connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -0,0 +1,275 @@
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.ORG_ID;
+import static org.apache.camel.LoggingLevel.INFO;
+
+@ApplicationScoped
+public class EmailRouteBuilder extends EngineToConnectorRouteBuilder {
+    @ConfigProperty(name = "user-provider.bop.url")
+    String bopURL;
+
+    @ConfigProperty(name = "user-provider.bop.api_token")
+    String bopApiToken;
+
+    @ConfigProperty(name = "user-provider.bop.client_id")
+    String bopClientId;
+
+    @ConfigProperty(name = "user-provider.bop.env")
+    String bopEnv;
+
+    @ConfigProperty(name = "user-provider.it.elements-per-page", defaultValue = "1000")
+    String itElementsPerPage;
+
+    @ConfigProperty(name = "user-provider.it.url")
+    String itUserServiceURL;
+
+    @ConfigProperty(name = "user-provider.rbac.url")
+    String rbacURL;
+
+    @ConfigProperty(name = "fetch.users.rbac.enabled", defaultValue = "true")
+    boolean fetchUsersWithRBAC;
+
+    @ConfigProperty(name = "user-provider.rbac.elements-per-page", defaultValue = "1000")
+    Integer rbacElementsPerPage;
+
+    @Inject
+    ConnectorConfig connectorConfig;
+
+    /**
+     * Processor which prepares the payload to be sent to BOP/MBOP, in order
+     * to trigger an email send.
+     */
+    @Inject
+    EmailPayloadPreparer emailPayloadPreparer;
+
+    /**
+     * Processor which takes the response from BOP/MBOP and transforms it to
+     * our internal users model.
+     */
+    @Inject
+    ITResponseProcessor itResponseProcessor;
+
+    /**
+     * Processor which builds the payload that BOP/MBOP expects.
+     */
+    @Inject
+    ITUserRequestPreparer itUserRequestPreparer;
+
+    /**
+     * Predicate which decides if we should keep looping for more user pages.
+     */
+    @Inject
+    NotFinishedFetchingAllPages notFinishedFetchingAllPages;
+
+    /**
+     * Processor which transforms the RBAC response into our internal users
+     * model.
+     */
+    @Inject
+    RBACUsersProcessor rbacUsersProcessor;
+
+    /**
+     * Processor which filters the users that should not be notified for a
+     * given event.
+     */
+    @Inject
+    RecipientsFilter recipientsFilter;
+
+    /**
+     * Template processor which takes the templates and the event and
+     * transforms them to a rendered string.
+     */
+    @Inject
+    TemplateProcessor templateProcessor;
+
+    /**
+     * Aggregator which aggregates all the users from the split into a single
+     * exchange property for later to be used.
+     */
+    @Inject
+    UsersAggregationStrategy usersAggregationStrategy;
+
+    /**
+     * Processor which extracts the "is the group platform default?" key from
+     * the received payload, to determine whether it is needed to fetch the
+     * principals from that group or just all the principals from the tenant.
+     */
+    @Inject
+    RBACGroupProcessor rbacGroupProcessor;
+
+    @Override
+    public void configureRoute() {
```

Comment:
It would be safer to include `removeHeaders("*")` before preparing each call to external services, to make sure headers are never leaked.

Reply:
Done. I need to test that I'm not removing all the necessary headers, and that specifying the URL parameters and what not via Camel headers isn't a problem.